app/models/link_history.rb

```diff
@@ -0,0 +1,24 @@
+class LinkHistory < ApplicationRecord
+  belongs_to :link
+
+  def add_errors(errors)
+    link_errors.keep_if { |link_error| errors.include?(link_error[:message]) }
```

Comment:
We can avoid doing this check and just leave `add_error` to do this.

Reply:
This line was removing any resolved errors from link_errors. We extracted it into a separate method for clarity.

Reply:
Oh I see, perhaps we should rename the method to `update_errors` to make it clearer that it doesn't just add errors.